src/lib/components/barchartv2/Barchart.component.tsx

```diff
@@ -25,24 +25,28 @@ export type TimeType = {
     interval: number;
   };
 };
-type Point = {
+export type Point = {
   key: string | number;
   values: { label: string; value: number }[];
 };
 
-export type BarchartProps = {
+export type BarchartProps<
+  T extends readonly {
+    readonly label: string;
+    readonly data: readonly (readonly [number | string, number | string])[];
+    readonly color: string;
+  }[],
```

Comment:
I'm unsure about this typing. I think the array needs to be converted to a tuple, but maybe readonly does that automatically now, do you have the sort typing as expected ?